### PR TITLE
If the frame handler thread is null do not schedule it on the executor

### PR DESCRIPTION
**Pull request by kannanjgithub**
Fixes NPE from "Detect transport executors with no remaining threads #11503".

### CODE REVIEW

**Review by ejona86 — COMMENTED** [review]
I think the latch/barrier handling needs to move into the try-catch that guarantees `clientFrameHandler` is set. So if the second thread fails, we'll use the source with "Use closed source on failure so that the reader immediately shuts down."

**Comment by kannanjgithub** [comment]
Only the source is guaranteed to be set after a certain point, so I think you mean source and not clientFrameHandler. I have pushed a commit with an attempt, although some other tests need fixing that fail with a NPE.

**Comment by ejona86** [comment]
Assigning `clientFrameHandler` is part of a `finally` block. https://github.com/grpc/grpc-java/blob/a431e3664be54a40dca12d62f81d22656f8d2057/okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java#L594

**Comment by kannanjgithub** [comment]
The case causing NPE for clientFrameHandler is [this latch.await()](https://github.com/grpc/grpc-java/blob/1612536f866645e000ea2d5825eb4560bc1a08c7/okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java#L523) timing out and sending a go away for internal error. How could this code be moved after clientFrameHandler is initialized in _finally_, because I thought we had to wait for connection preface settings to be [sent first](https://github.com/grpc/grpc-java/blob/1612536f866645e000ea2d5825eb4560bc1a08c7/okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java#L622).

**Comment by ejona86** [comment]
Move it into the try block, so if it fails `clientFrameHandler` will be initialized still via the finally.

**Comment by kannanjgithub** [comment]
Got it, I wrongly assumed you were implying changing the order of the statements and got confused, you were only suggesting expanding the try scope to include the latch and barrier await. Pushed the commit.

File: okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java [hunk 1]
```diff
@@ -587,6 +583,11 @@ sslSocketFactory, hostnameVerifier, sock, getOverridenHost(), getOverridenPort()
         } catch (StatusException e) {
           startGoAway(0, ErrorCode.INTERNAL_ERROR, e.getStatus());
           return;
+        } catch (TimeoutException | BrokenBarrierException e) {
```
Comment (ejona86) [inline okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java:586]: This needs to be a catch only for latch.await/barrier.await.

**Review by ejona86 — APPROVED** [review]

**Pull request merged**